[BUG] mypy type checking fails: PandasDataFrameLike missing 'data' attribute and instance-only attribute access errors in umap.py and test_random_forest.py

**Describe the bug**
Build: spark-rapids-ml_nightly/993

The mypy type checking step failed with 5 errors in 2 files during the nightly CI run. The errors are: (1) `PandasDataFrameLike` has no attribute `data` in umap.py, and (2) instance-only attributes `dtype` and `n_cols` accessed on class objects in test_random_forest.py. The build fails at the "Type checking..." phase in `run_test.sh`.

Error logs:
```
src/spark_rapids_ml/umap.py:1705: error: PandasDataFrameLike? has no attribute "data"  [attr-defined]
tests/test_random_forest.py:338: error: Cannot access instance-only attribute "dtype" on class object  [misc]
tests/test_random_forest.py:340: error: Cannot access instance-only attribute "dtype" on class object  [misc]
tests/test_random_forest.py:341: error: Cannot access instance-only attribute "n_cols" on class object  [misc]
tests/test_random_forest.py:342: error: Cannot access instance-only attribute "n_cols" on class object  [misc]
Found 5 errors in 2 files (checked 67 source files)
```

**Environment details**
- Python 3.10
- mypy 1.20.0 (upgraded via requirements_dev.txt change: `mypy>=1.0.0`)
- CUDA 13.1, NVIDIA A100-PCIE-40GB
- pyspark 3.4.4

--- a/python/requirements_dev.txt
+++ b/python/requirements_dev.txt
@@ -16,7 +16,7 @@
 black==25.12.0
 build>=0.10.0
 isort==7.0.0
-mypy>=1.0.0
+mypy==1.19.1
 numpydoc
 pydata-sphinx-theme
 pylint

